src/Context/FilterContext.tsx

```diff
@@ -0,0 +1,55 @@
+import { FC, createContext, useContext, useMemo, useState } from 'react';
+import { TodoContext } from './TodoContext';
+
+export interface IFilterContext {
+  filterType: string;
+  filteredTodos: [];
+  showAllTodos: () => void;
+  showCompletedTodos: () => void;
+  showActiveTodos: () => void;
+}
+
+const initialState: IFilterContext = {
+  filterType: 'All',
+  filteredTodos: [],
+  showAllTodos: () => {},
+  showCompletedTodos: () => {},
+  showActiveTodos: () => {},
+};
+
+export const FilterContext = createContext(initialState);
+
+type TProps = {
+  children: React.ReactNode;
+};
+
+export const FilterProvider: FC<TProps> = ({ children }) => {
+  const { todos } = useContext(TodoContext);
+
+  const [filterType, setFilterType] = useState<string>('All');
+
+  const filteredTodos = useMemo(() => {
+    if (filterType === 'Completed') {
+      return todos.filter(todo => todo.completed);
+    } else if (filterType === 'Active') {
+      return todos.filter(todo => !todo.completed);
+    } else {
+      return todos;
+    }
```

Comment:
better to you switch/case here, 'cause you check different values of filterType variable